daemon/restart.go

```diff
@@ -35,11 +36,23 @@ func (daemon *Daemon) containerRestart(container *container.Container, seconds i
 		defer daemon.Unmount(container)
 	}
 
-	if err := daemon.containerStop(container, seconds); err != nil {
+	// set AutoRemove flag to false before stop so the container won't be
+	// removed during restart process
+	autoRemove := container.HostConfig.AutoRemove
+
+	container.HostConfig.AutoRemove = false
+	err := daemon.containerStop(container, seconds)
+	// restore AutoRemove irrespective of whether the stop worked or not
+	container.HostConfig.AutoRemove = autoRemove
+	if toDiskErr := container.ToDiskLocking(); toDiskErr != nil {
+		logrus.Errorf("Write container to disk error: %v", toDiskErr)
+	}
```

Comment:
let's add a comment in the code here (and above in the Shutdown() func) so people know what's going on. They won't know to look at this PR for that info.

Also, should this call to ToDiskLocking() only be done upon an error?